packages/core/src/utils-hoist/syncpromise.ts

```diff
@@ -137,59 +189,4 @@ class SyncPromise<T> implements PromiseLike<T> {
       });
     });
   }
-
-  /** JSDoc */
-  private readonly _resolve = (value?: T | PromiseLike<T> | null) => {
```

Comment:
l: Instead of defining these fields in the constructor, can we declare them as regular methods (i.e. without the assignment)? I might be missing something as to why we didn't do this before though

So basically, like the public methods above, just as private? (Which I'm aware doesn't prevent them from being called externally but that should be fine)

Reply:
we need to do `bind` then a bunch of times, which... is also possible probably, let me check the bundle size diff!

Reply:
I rewrote this here to generally be a bit cleaner, so this should make more sense now!